provider/pkg/await/awaiters.go

```diff
@@ -689,7 +689,7 @@ func untilCoreV1ServiceAccountInitialized(c awaitConfig) error {
 	// k8s v1.24 changed the default secret provisioning behavior for ServiceAccount resources, so don't wait for
 	// clusters >= v1.24 to provision a secret before marking the resource as ready.
 	// https://github.com/kubernetes/kubernetes/blob/v1.24.3/CHANGELOG/CHANGELOG-1.24.md#urgent-upgrade-notes
-	if c.clusterVersion.Compare(cluster.ServerVersion{Major: 1, Minor: 24}) >= 0 {
+	if c.clusterVersion != nil && c.clusterVersion.Compare(cluster.ServerVersion{Major: 1, Minor: 24}) >= 0 {
```

Comment:
I don't see a problem with taking the assumption that `c.clusterVersion` is non-nil. An assertion would make more sense than conditional logic, unless there's actually a use-case where cluster version is unknown.
```go
import "github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
```